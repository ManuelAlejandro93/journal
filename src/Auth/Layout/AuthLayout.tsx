import { Grid, Typography } from '@mui/material';

interface Props {
  authPageName: string;
  children: React.ReactNode;
}

export const AuthLayout = ({ authPageName, children }: Props) => {
  return (
    <Grid
      container
      spacing={0}
      direction={'column'}
      alignItems={'center'}
      justifyContent={'center'}
      sx={{ minHeight: '100vh', backgroundColor: 'primary.main', padding: 4 }}
    >
      <div className='grid shadow-2xl shadow-black grid-cols-2 rounded-md bg-white p-4 space-x-4'>
        <div className='col-span-2 text-center'>
          <Typography fontSize={'2rem'}>
            {authPageName.toUpperCase()}
          </Typography>
        </div>
        {children}
      </div>
    </Grid>
  );
};
