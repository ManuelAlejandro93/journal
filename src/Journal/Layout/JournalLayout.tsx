import { Navbar, SideBar } from '@/Journal';
// import { Toolbar } from '@mui/material';

interface Props {
  children: React.ReactElement[] | React.ReactElement;
}

export const JournalLayout = ({ children }: Props) => {
  return (
    <div className='flex animate-jump-in animate-duration-500'>
      <SideBar />
      <Navbar />
      <main className='p-4 mt-20'>{children}</main>
    </div>
  );
};
