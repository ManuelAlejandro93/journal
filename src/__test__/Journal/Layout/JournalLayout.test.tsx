import { render, screen } from '@testing-library/react';
import { JournalLayout } from '../../../Journal/Layout/JournalLayout';

describe('Pruebas en <JournalLayout/>', () => {
  test('Debe ser renderizable', () => {
    screen;
    const { container } = render(
      <JournalLayout>
        <></>
      </JournalLayout>
    );
    expect(container).toBeTruthy();
  });

  test('Debe ser igual al snapshot', () => {
    const { container } = render(
      <JournalLayout>
        <></>
      </JournalLayout>
    );
    expect(container).toMatchSnapshot();
  });
});
