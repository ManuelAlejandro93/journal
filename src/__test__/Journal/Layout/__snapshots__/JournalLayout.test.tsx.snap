// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalLayout/> Debe ser igual al snapshot 1`] = `
<div>
  <div
    class="flex animate-jump-in animate-duration-500"
  >
    <main
      class="p-4 mt-20"
    />
  </div>
</div>
`;
